lib/roo_on_rails/concerns/require_api_key.rb

```diff
@@ -83,7 +83,7 @@ def initialize
         end
 
         def valid?(service_name, client_key)
-          return false if service_name == '' || client_key == ''
+          return false unless service_name && client_key
```

Comment:
What do you think about `if service_name.to_s == '' || client_key.to_s == ''` ? `unless` with `&&` is always harder to follow for me 😵

Reply:
I'd personally avoid the cost of additional `.to_s` -- and quite frankly find `unless` more readable 😄 

But don't feel strong about it, if that's the general preference, I'm happy to change

Reply:
If we're worried about the cost of `to_s` then we should be coding in Assembler, not in Ruby 😉 

What you've written so far doesn't actually cover both cases though; we need to duplicate `blank?` functionality — it needs to return `false` if either are nil or empty string (which is also covered in Florian's comment).

Maybe: `if service_name.to_s.empty? || client_key.to_s.empty?`?

Reply:
I understand the edge case -- I don't see a test case covering it tho! 😉

Reply:
BTW, in this case we don't need to catch the whitespace here, given it will then continue and fail finding a valid key for `''` -- but I made the change anyway :)